<?xml version="1.0"?>

<launch>

<!-- launch file to kickstart estimation, mapping, navigation -->

<!-- CONFIGURATION  -->
  
  <arg name="config_okvis" value="$(find ros_submapping)/okvis2/config/config_uhumans_Tommaso.yaml" /> <!-- configuration file -->
  <arg name="package" value="$(find ros_submapping)"/> <!-- to launch slam or simple example -->
  <arg name="config_s8" value="$(find ros_submapping)/okvis2/config/config_uhumans_depth_Tommaso.yaml" /> <!-- configuration file -->
  <arg name="rviz" default="true"/>

  <group ns="submapping">
    
    <node name="main" pkg="ros_submapping" type="main" args="$(arg config_okvis) $(arg package) $(arg config_s8)" output="screen" /> <!-- launch-prefix="xterm -e gdb - - args" -->
    
  </group>
  
  <group if="$(arg rviz)">
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find rviz_rotors)/rviz/config_1.rviz" output="screen" launch-prefix="xterm -e"/>
  </group>

  <!-- starting location. to make estimates consistent with ground truth -->
  <node pkg="tf" type="static_transform_publisher" name="link1_broadcaster" args="-16.04391 -2.61944 2.461393 0 0 -0.04641965028571606 0.998922027020804 world odom 100" />

</launch>
